Sources/Dotenv.swift

```diff
@@ -158,15 +162,29 @@ public enum Dotenv {
         set(value: value?.stringValue, forKey: key, overwrite: overwrite)
     }
 
+#if os(Linux)
     /// Set a value in the environment.
     /// - Parameters:
     ///   - value: Value to set.
     ///   - key: Key to set the value with.
     ///   - overwrite: Flag that indicates if any existing value should be overwritten, defaults to `true`.
     public static func set(value: String?, forKey key: String, overwrite: Bool = true) {
+        guard let value = value else  {
+            setenv(key, "", overwrite ? 1 : 0)
```

Comment:
With a nil value it won't compile, but more specifically here's some docs

> The setenv function can be used to add a new definition to the environment. The entry with the name name is replaced by the value ‘name=value’. Please note that this is also true if value is the empty string. To do this a new string is created and the strings name and value are copied. A null pointer for the value parameter is illegal. If the environment already contains an entry with key name the replace parameter controls the action. If replace is zero, nothing happens. Otherwise the old entry is replaced by the new one. 

[setenv man page](https://man7.org/linux/man-pages/man3/setenv.3.html)

I was just rushing to get it to work, but after reading the docs I don't know if empty string is the correct approach. Given that it evaluates to true.

Reply:
Reading the source adding another guard in the Value init? So if the stringValue is empty it returns nil should work, but I could be missing an expected behavior.

![stringnil](https://github.com/thebarndog/swift-dotenv/assets/3574726/dc740ce3-0299-477b-8a82-c8dfa8894300)